packages/client-react-streaming/src/DataTransportAbstraction/WrappedApolloClient.tsx

```diff
@@ -26,13 +27,24 @@ type SimulatedQueryInfo = {
   options: WatchQueryOptions<OperationVariables, any>;
 };
 
-class ApolloClientSSRImpl<TCacheShape> extends ApolloClient<TCacheShape> {
-  watchQueryQueue = createBackpressuredCallback<WatchQueryOptions<any>>();
-
+const wrappers = Symbol.for("apollo.hook.wrappers");
+class ApolloClientBase<TCacheShape> extends ApolloClient<TCacheShape> {
   constructor(options: ApolloClientOptions<TCacheShape>) {
     super(options);
+
+    if (!(this.cache instanceof WrappedInMemoryCache)) {
+      throw new Error(
+        "When using Apollo Client streaming SSR, you must use the `InMemoryCache` variant provided by the streaming package."
+      );
+    }
   }
 
+  [wrappers] = hookWrappers;
```

Comment:
See https://github.com/apollographql/apollo-client/pull/11617/files#r1500875764

I'd be curious if we could store this on `QueryManager` which would make a path for `useReadQuery` to work in all scenarios.